### ISSUE TITLE
Incorrect HTTP status handling in uid2::UID2Client::Refresh()

### ISSUE THREAD

**Issue by spiritloose** [IABTechLab/uid2-client-cpp11#3]
The `uid2::UID2Client::Refresh()` method will attempt to parse JSON even in the case of HTTP errors.

This causes, for example, `uid2::RefreshResult::GetReason()` function returns  `"expected value, got 'U' (85)"` when 401 unauthorized.

A patch to correct this problem is below.

```diff
diff --git a/lib/uid2client.cpp b/lib/uid2client.cpp
index 70c0b55..eaef82a 100644
--- a/lib/uid2client.cpp
+++ b/lib/uid2client.cpp
@@ -156,7 +156,7 @@ namespace uid2
        {
                if (auto res = httpClient.Get("/v1/key/latest"))
                {
-                       if (res->status >= 200 || res->status < 300)
+                       if (res->status >= 200 && res->status < 300)
                        {
                                return res->body;
                        }
```

**Comment by spiritloose** [comment]
Starting with commit 45960ea44ecb44abf5ee40aca267a05f520b9966, an exception is now thrown on HTTP error.

```
# uname -a
Linux c3ba9c947eb3 5.10.104-linuxkit #1 SMP Wed Mar 9 19:05:23 UTC 2022 x86_64 GNU/Linux
# ./app/example https://integ.uidapi.com `head -c 24 /dev/urandom | base64` `head -c 24 /dev/urandom | base64` foobarbaz

EXAMPLE: Basic keys refresh and decrypt token

terminate called after throwing an instance of 'char const*'
Aborted
```

It seems to be trying to base64 decode the text `{"status":"unauthorized"}` and failing with the error message `invalid base64 string length`.

**Comment by spiritloose** [comment]
Fixed by 150deecfe2d9595118e102ec793c1e096238ced0. Thanks.
